getting_started/scripting/gdscript/gdscript_basics.rst

```diff
@@ -1463,63 +1463,114 @@ freed.
 
 Signals
 ~~~~~~~
+It is often desired to send a notification from an instance to another
+in a way that does not require the notifier knowing about the target of
+the notification. GDScript's mechanism is called 'signals' and custom
+signals are easily created using the `signal` keyword.
 
-It is often desired to send a notification that something happened in an
-instance. GDScript supports creation of built-in Godot signals.
-Declaring a signal in GDScript is easy using the `signal` keyword.
+::
+
+   # Signal with no arguments
+   signal your_signal_name
+
+   # Signal with two arguments
+   signal your_signal_name_with_args(a, b)
+
+These signals are used by connecting them to signal handlers commonly
+referred to as slots. Connecting custom signals to slots can be done
+in the editor or in GDScript just like the built-in signals found in
+classes like Button or RigidBody.
+
+Here's an example of creating a custom signal in one script and then
+connecting it to a custom slot in a separate script:
+
+::
+
+   your_notifier.gd
+   ----------------
+
+   signal your_notify
```

Comment:
It would be preferable if example code stuck to Godot naming conventions. Godot built-in signals use past-tense verb nomenclature: `body_entered`, `finished`, etc.

Reply:
Changed in all the examples with f8290126